libflux/go/libflux/buffer.go

```diff
@@ -0,0 +1,45 @@
+package libflux
+
+// #cgo CFLAGS: -I.
+// #cgo LDFLAGS: -L. -lflux
+// #include "flux.h"
+// #include <stdlib.h>
+import "C"
+
+import (
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	flatbuffers "github.com/google/flatbuffers/go"
+)
+
+type ManagedBuffer struct {
+	Buffer []byte
+	Offset flatbuffers.UOffsetT
+	freeFn func()
+}
+
+func (b *ManagedBuffer) Free() {
+	if b.freeFn != nil {
+		b.freeFn()
+		b.freeFn = nil
+	}
+}
+
+func unwrapBuffer(buf C.struct_flux_buffer_t) *ManagedBuffer {
+	sh := new(reflect.SliceHeader)
+	sh.Data = uintptr(unsafe.Pointer(buf.data))
+	sh.Len = int(C.uint(buf.len))
+	sh.Cap = sh.Len
+	bs := *(*[]byte)(unsafe.Pointer(sh))
+	ret := &ManagedBuffer{
+		Buffer: bs,
+		Offset: flatbuffers.UOffsetT(uintptr(C.uint(buf.offset))),
+		freeFn: func() {
```

Comment:
I wonder if this can just be a regular method on `ManagedBuffer`, that will call `C.flux_free` on `buf.data` and then set `buf.data` to zero?  That seems simpler and easier to follow.